### PR TITLE
Split travis runs into stages / only run unit tests for PRs

### PR DESCRIPTION
**Pull request by ulope**
Only run `smoketest` and unit tests for PRs by default.
Nightly cron runs as well as tagged deploy commits also run integration tests.

Override with `[ci integration]` in the commit message of the `HEAD` commit of a PR.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,60 +40,109 @@ env:
...
+      env:
+        - TEST='raiden/tests/unit'
+
+    - stage: integration
+      <<: *job-template-integration
+      env:
+        - TEST='raiden/tests/integration --ignore=raiden/tests/integration/transfer --ignore=raiden/tests/integration/long_running --ignore=raiden/tests/integration/api --ignore=raiden/tests/integration/contracts'
+        - TRANSPORT_OPTIONS='--transport=udp'
+
+#    - stage: integration
```
Comment (LefterisJP) [inline .travis.yml]: left commented out code in here, also below. Are these the skipped matrix tests?

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,34 @@
...
+      - export PATH=$PATH:$HOME/.bin
+      - .travis/before_install.sh
+    install:
+      - .travis/install.sh
+    script:
+      - .travis/run.sh
+    after_script:
+      - coveralls
+
+  job-template-integration: &job-template-integration
```
Comment (LefterisJP) [inline .travis.yml]: question: So a job template is just all the steps you have in the template. And this particular one is just like the `job-template-test` but with sudo enabled full VMs

Comment (ulope) [reply .travis.yml]: Yes, this a taking advantage of [YAML anchors](https://blog.daemonl.com/2016/02/yaml.html).
It's just expanding the 'template' into the job definitions and in this case adding the `sudo` setting.

File: .travis.yml (hunk 1 above)
Comment (ulope) [reply .travis.yml]: Exactly.

**Pull request merged**